### PR TITLE
Keep C# enum value sequence

### LINKED ISSUE #29: Enum offsets aren't converted correctly
Hey there,

I think I found a bug.

Consider the following C#:
```csharp
public enum BillingCycle
{
   Yearly = 7,
   Monthly,
   Never = 4
}
```

If no explicit value is defined for an enum field, the C# compiler will increment the previous field value by +1. The C# compiler actually generates the above enum as:
```csharp
public enum BillingCycle
{
    Yearly = 7,
    Monthly = 8,
    Never = 4
}
```
You can test that here: https://sharplab.io/#v2:CYLg1APgAgzABAUwHYFcC2cBCBLANr7JAcwGEBPAY1wQFgAoAb3rhYE0EBDAJ1zLgF44AdgA0zFgFkA9kgAuAC15i6LOADkEANwRcBcACz0AvkA=

The `csharp-models-to-typescript` will generate the incorrect enum mapping as follows:
```csharp
export enum BillingCycle {
    Yearly = 7,
    Monthly = 1,
    Never = 4,
}
```

Version Info:
"csharp-models-to-typescript": "^0.16.0"

### PR DESCRIPTION
**Pull request by digocesar**
As I could test, TypeScript folow the same C# numbering sequence when enum value is missing in declaring type.
I think that if no number is defined in C#, it could remains not defined in TypeScript.
@svenheden and @SafeerH, any idea of problems we might have with this change?

**C#:** https://sharplab.io/#v2:CYLg1APgAgzABAUwHYFcC2cBCBLANr7JAcwGEBPAY1wQFgAoAb3rhYEEAaZlzOAXjgDsnOizglhogCJ84AFgksAovQC+QA==
**TypeScript:** https://www.typescriptlang.org/play/?#code/KYOwrgtgBAQglgGwXEBzAwgTwMYOAKAG98pSBBAGhNJigF4oB2K0qdF0gEXqgBYOoAUXwBfIA

Closes #29

### CODE REVIEW

File: converter.js [hunk 1]
```diff
@@ -113,9 +113,13 @@ const createConverter = config => {
...
             rows.push(`export enum ${enum_.Identifier} {`);
 
-            entries.forEach(([key, value], i) => {
+            entries.forEach(([key, value]) => {
                 if (config.numericEnums) {
-                    rows.push(`    ${key} = ${value != null ? value : i},`);
+                    if (value == null) {
+                        rows.push(`    ${key},`);
+                    } else {
+                        rows.push(`    ${key} = ${value != null ? value : i},`);
```
Comment (svenheden) [inline converter.js:121]: `value` is always `!= null` here, so this ternary expression can be removed

**Comment by svenheden** [comment]
> As I could test, TypeScript folow the same C# numbering sequence when enum value is missing in declaring type.
> I think that if no number is defined in C#, it could remains not defined in TypeScript.
> @svenheden and @SafeerH, any idea of problems we might have with this change?

No to my ears that sounds solid, go for it 👍

File: converter.js (hunk 1 above)
Comment (digocesar) [reply converter.js]: Thats right! My mistake!
I did an ammend for previous commit.

**Comment by SafeerH** [comment]
> As I could test, TypeScript folow the same C# numbering sequence when enum value is missing in declaring type. I think that if no number is defined in C#, it could remains not defined in TypeScript. @svenheden and @SafeerH, any idea of problems we might have with this change?
> 
> **C#:** https://sharplab.io/#v2:CYLg1APgAgzABAUwHYFcC2cBCBLANr7JAcwGEBPAY1wQFgAoAb3rhYEEAaZlzOAXjgDsnOizglhogCJ84AFgksAovQC+QA== **TypeScript:** https://www.typescriptlang.org/play/?#code/KYOwrgtgBAQglgGwXEBzAwgTwMYOAKAG98pSBBAGhNJigF4oB2K0qdF0gEXqgBYOoAUXwBfIA
> 
> Closes #29

Sounds good to me! 👍

**Comment by SafeerH** [comment]
@digocesar could you add `G` value to the TestEnum in TestClass.cs just for the completeness?
```cs
public enum TestEnum {
    ...
    G    // 27 in decimal
}
```
Also, some suggestions to consider:
- Change the `TestClass.cs` file name to `TestFile.cs` (because we now have enum as well)
- Change the namespace to `TestNamespace` (instead of `TestClass`)

**Comment by digocesar** [comment]
> @digocesar could you add `G` value to the TestEnum in TestClass.cs just for the completeness?
> 
> ```cs
> public enum TestEnum {
>     ...
>     G    // 27 in decimal
> }
> ```
> 
> Also, some suggestions to consider:
> 
> * Change the `TestClass.cs` file name to `TestFile.cs` (because we now have enum as well)
> * Change the namespace to `TestNamespace` (instead of `TestClass`)

Done.

**Comment by SafeerH** [comment]
Hey guys @svenheden @digocesar, I think it is a high time we publish a new NPM release. Also, what do you guys think about bumping up the major version - because we migrated to .NET 8 and we have a lot of other good improvements as well?

**Comment by digocesar** [comment]
> Hey guys @svenheden @digocesar, I think it is a high time we publish a new NPM release. Also, what do you guys think about bumping up the major version - because we migrated to .NET 8 and we have a lot of other good improvements as well?

I was thinking to suggest it too.
I think we should release a version so that the component works again and then we will make other improvements.

**Comment by svenheden** [comment]
Sounds good guys! I just bumped the version to 1.0.0 and published to npm.

**Pull request merged**